LinqGen.Generator/Instructions/Evaluations/Local/SingleEvaluation.cs

```diff
@@ -0,0 +1,48 @@
+﻿namespace Cathei.LinqGen.Generator;
+
+public sealed class SingleEvaluation : LocalEvaluation {
+    private bool OrDefault { get; }
+
+    public SingleEvaluation(in LinqGenExpression expression, uint id, bool orDefault) : base(expression, id)
+    {
+        OrDefault = orDefault;
+    }
+
+    protected override TypeSyntax ReturnType => Upstream.OutputElementType;
```

Comment:
Minor comment that we could add this line since `Single` consumes up to 2 elements. This gives hint to operations, for example to `OrderBy` that it doesn't have to sort rest of the elements other than top 2.
```cs
    protected override ExpressionSyntax TakeExpression => LiteralExpression(2);
```